spyder/plugins/pylint/widgets/pylintgui.py

```diff
@@ -244,6 +242,14 @@ def __init__(self, parent, max_entries=100, options_button=None,
         else:
             self.start_button.setEnabled(False)
 
+        if self.parent:
+            self.curr_filenames = self.parent.get_option('history_filenames')

```

Comment:
Please change this to

    self.parent.get_option('history_filenames', [])

So that it uses an empty list the first time this option is called.